README.rst

```diff
@@ -8,17 +8,33 @@ directly to your Python code based on the types collected at runtime.
 Examples
 --------
 
+In the example, we have two files: ``some/module.py`` and ``myscript.py``.
+
+Say ``some/module.py`` originally contained::
+
+  def add(a, b):
+      return a + b
+
+And ``myscript.py`` contains::
+
+  from some.module import add
+
+  add(1, 2)
+
+Now we want to infer the type annotation of ``add`` in ``some/module.py`` by
+running ``myscript.py`` with ``MonkeyType``. This is supported in two ways:
+
 Run a script under call-trace logging of functions and methods in all imported
 modules::
 
   $ monkeytype run myscript.py
 
-Or enable call-trace logging for a block of code::
+Or enable call-trace logging for a block of code in ``myscript.py``::
```

Comment:
I'm slightly concerned that this gives the wrong impression that the context manager is only for use in a script. The main reason the context manager exists is that in some cases you might want to run tracing on some code that isn't a script at all. I think it might be better to move the context manager down to after the main example, letting the primary example flow more smoothly with only `monkeytype run`.

Reply:
You mean that it can also be used in an interactive shell?

 I actually feel this example is a bit too long as well.